core/pkg/service/sync/sync.go

```diff
@@ -0,0 +1,139 @@
+package sync
```

Comment:
I see several components here - a core similar to a `runtime`, `subscription handler` & `grpc/connect listener`. A refactoring here with good contracts will help with testing and maintenance. 

For example, `rpc.NewFlagSyncServiceHandler(s)`, parsing `SyncServer` as `s` is a little misleading.

Reply:
> For example, rpc.NewFlagSyncServiceHandler(s), parsing SyncServer as s is a little misleading.

This implementation is copied from `runtime`, happy to create an issue to refactor both, as they should follow a similar pattern

Reply:
Happy to refactor this, could you elaborate on the structure you are suggesting?